src/arduino.c

```diff
@@ -131,9 +170,12 @@ void arduino_initpgm(PROGRAMMER *pgm) {
   stk500_initpgm(pgm);
 
   strcpy(pgm->type, "Arduino");
+  PDATA(&pgm)->autoreset = true; // Auto-reset enabled by default
```

Comment:
This looks wrong (because it probably is): *Must* be `PDATA(pgm)` not `PDATA(&pgm)`.I think this sets an unknown memory address to 1 (but not `autoreset`).

Reply:
It works this way, and I got a segfault if not. I'll look into it again

Reply:
I think I know what happens: `arduino_initpgm()` is called *before* `pgm->setup()` which allocates the pdata structure. So, `pgm->cookie` is NULL here (as it hasn't been initialises yet) and you get a seg-fault when you access it. `PDATA(&pgm)` is a memory location in the PROGRAMMER sructure which is at the offset of autoreset from where the cookie pointer lives. This is defo not what you want! *Either* set `PDATA(pgm)->autoreset` in `pgm->setup()` or later but before the parameters are parsed. *Or* invert the logic: Use a variable `noautoreset` that gets automatically initialised to zero. So you don't need to initialise that.